flang/lib/Evaluate/intrinsics.cpp

```diff
@@ -434,6 +435,9 @@ static const IntrinsicInterface genericIntrinsicFunction[]{
             {"shift", AnyInt, Rank::dimRemovedOrScalar}, OptionalDIM},
         SameType, Rank::conformable, IntrinsicClass::transformationalFunction},
     {"dble", {{"a", AnyNumeric, Rank::elementalOrBOZ}}, DoublePrecision},
+    {"derf", {{"x", DoublePrecision}}, DoublePrecision},
```

Comment:
Your new functions are not generic intrinsic functions and do not belong in the generic intrinsic function table.  Use the specific intrinsic function table (`specificIntrinsicFunction`) below to define these cases as special cases of `ERF`, &c.